File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        freq = {}
        n = len(nums)
        for i in nums:
            if i in freq:
                freq[i] += 1
            elif i not in freq:
                freq[i] = 1
            
            if freq[i] > n/2:
                return i

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int n = nums.length;
        int count = 0, len = 0;
        HashSet<Integer> set = new HashSet<>();
        for(int num: nums) {
            set.add(num);
        }
        for(int num: nums) {
            int prevElement = num - 1;
            if(!set.contains(prevElement)) {
                len = 1;
                int nextElement = num + 1;
                while(set.contains(nextElement)) {
                    len++;
                    nextElement++;
                }
            }
            count = Math.max(len, count);
        }
        return count;
    }
}
